### ISSUE TITLE
Cases handling on "Not enough inputs!"

### ISSUE THREAD

**Issue by Kit-Law-42** [se2p/ps-wise-2021-program-description#35]
Hi, I am a bit confused with the case "Not enough inputs".
How many results should we print out if we only have "1 2 R" as in put in basic-test.js?
Would it be like below, just as we type "1 2 R" and then Ctrl+D / Ctrl+C to manually end a debug session in shell?
```
┌───┬───┬───┐[\n]
│   │ 1 │   │[\n]
├───┼───┼───┤[\n]
│   │   │   │[\n]
├───┼───┼───┤[\n]
│   │   │   │[\n]
└───┴───┴───┘[\n]
╔═══════════╗[\n]
║           ║[\n]
╚═══════════╝[\n]

>┌───┬───┬───┐[\n]
│   │ 1 │   │[\n]
├───┼───┼───┤[\n]
│   │   │   │[\n]
├───┼───┼───┤[\n]
│   │   │   │[\n]
└───┴───┴───┘[\n]
╔══════════════════╗[\n]
║Not enough inputs!║[\n]
╚══════════════════╝[\n]
```
Or there would be only 1 chessboard showing there is not enough inputs, just like the 2nd board above? 

If it is the latter case, I think the behavior would be different in using mocha and testing manually? Since in the shell we are always awaiting users for additional inputs.

**Comment by lechlm** [comment]
In my opinion, the first case makes more sense because first the program processes the input "1 2 R" and outputs the new UI state, as described in the program description. And afterwards, the program "sees" that there is no further input and now ends the game with the final board state + the message. In the latter case, two inputs ("1 2 R" and "end of input stream") would be treated as a single event, not one after the other, which does not reflect the way how the inputs are processed in general.

**Comment by alessiogambi** [comment]
Hi, the expected behavior is the following:
- process the input, refresh the GUI, shows the console (`>`)
- realize there's no more input to elaborate, refresh the GUI, print the "not enough input" message, terminate without showing the console (`>`)

Assuming that there are enough inputs, this is conceptually similar to:
- process the input, refresh the GUI, shows the console (`>`)
- process the (last) input, realize the game is over, refresh the GUI, print the win/lost message, terminate without showing the console (`>`)

Basically, there is "no prefetching" of the inputs to see how many are left, simply loop over the inputs until there's something and exit otherwise.

Regarding manual testing, you should be able to simulate the "not enough inputs" case as follow:

```
echo -e "1 1 R" | java Minesweeper simple.cfg 
echo -e "1 1 R" | node minesweeper.js simple.cfg 
```

**Comment by Kit-Law-42** [comment]
Hi! Thanks for the explanation.
Everything is working on WSL right now.
